packages/core/src/server.ts

```diff
@@ -1174,6 +1179,78 @@ export class Controller extends API {
         });
     }
 
+    /**
+     * Makes an API request to HIBP to check whether the user's password has been detected in any known data breaches
+     * Returns a Promise containing an PasswordBreachResult.
+     * @param sha1Hash Hash of user's password.
+     */
+    async getPasswordBreachStatus({ sha1Hash }: GetPasswordBreachParams) {
+        this._requireAuth();
+
+        this.log("passwordBreachStatus.get", {
+            passwordHash: sha1Hash
+        });
+
+        return new Promise<PasswordBreachResult>((resolve, reject) => {
+            const sha1HashPrefix = sha1Hash.substring(0,5);
+            const sha1HashSuffix = sha1Hash.substring(5);
+            const url = `https://api.pwnedpasswords.com/range/${sha1HashPrefix}`;
+
+            axios.get(url)
```

Comment:
Please don't import an external packages just for making network requests. We're trying to keep external dependencies to an absolute minimum and the core package in particular should stay completely dependency-free.